File: advent-of-code/2015/02/02/solution.py
from pathlib import Path


def calculate_ribbon(dimensions):
    """Calculate ribbon needed for a present."""
    dims = sorted(map(int, dimensions.split('x')))
    
    # Smallest perimeter (using two smallest dimensions)
    perimeter = 2 * (dims[0] + dims[1])
    
    # Volume for bow
    volume = dims[0] * dims[1] * dims[2]
    
    return perimeter + volume

# Test cases with assertions
assert calculate_ribbon("2x3x4") == 34
assert calculate_ribbon("1x1x10") == 14

# Read input and calculate total
input_file = Path(__file__).parent.parent / "input.txt"
total = sum(calculate_ribbon(line.strip()) for line in input_file.read_text().splitlines() if line.strip())
print(f"Python: {total}")

File: advent-of-code/2015/02/02/solution.cs
using System.Diagnostics;

static int CalculateRibbon(string dimensions)
{
    var dims = dimensions.Split('x').Select(int.Parse).OrderBy(x => x).ToArray();
    
    // Smallest perimeter (using two smallest dimensions)
    var perimeter = 2 * (dims[0] + dims[1]);
    
    // Volume for bow
    var volume = dims[0] * dims[1] * dims[2];
    
    return perimeter + volume;
}

// Test cases with assertions
Debug.Assert(CalculateRibbon("2x3x4") == 34);
Debug.Assert(CalculateRibbon("1x1x10") == 14);

// Read input and calculate total
var scriptDir = Directory.GetCurrentDirectory();
var inputPath = Path.Combine(scriptDir, "..", "input.txt");
var total = File.ReadAllLines(inputPath)
                .Where(line => !string.IsNullOrWhiteSpace(line))
                .Sum(CalculateRibbon);

Console.WriteLine($"C#: {total}");
